src/main/java/st/ratpack/auth/CachingTokenValidator.java

```diff
@@ -0,0 +1,37 @@
+package st.ratpack.auth;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import ratpack.exec.Promise;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public class CachingTokenValidator implements TokenValidator {
+
+	public static long DEFAULT_TTL = 5000;
+
+	private final LoadingCache<String, Promise<Optional<OAuthToken>>> cache;
+
+	public CachingTokenValidator(AuthModule.Config config, final TokenValidator tokenValidator) {
+		Long ttl = config.ttl != null ? config.ttl : DEFAULT_TTL;
+		System.out.println(tokenValidator);
+
+		cache = CacheBuilder.newBuilder()
+				.expireAfterWrite(ttl, TimeUnit.MILLISECONDS)
+				.build(new CacheLoader<String, Promise<Optional<OAuthToken>>>() {
+
+					@Override
+					public Promise<Optional<OAuthToken>> load(String key) throws Exception {
+						System.out.println(tokenValidator);
+						return tokenValidator.validate(key).cache();
```

Comment:
This is calling the internal cache to the Ratpack Promise and the cache this guava cache so its doubling up what is being stored.